kbase-extension/static/kbase/js/widgets/function_output/kbaseBlastOutput.js

```diff
@@ -1,430 +1,459 @@
+                            start = i;
+                            end = i + 60;
+                            var p1 = hseq.substring(start, end);
+                            var p2 = midline.substring(start, end);
+                            var p3 = qseq.substring(start, end);
+
+                            if (i == 0) {
+                                q_start = Number(query_from);
+                                h_start = Number(hit_from);
+                            }
+                            else {
+                                h_start = h_end + 1;
+                                q_start = q_end + 1;
+                            }
+
+                            var c1 = p1.replace(/-/g, "");
+                            var c3 = p3.replace(/-/g, "");
+
+                            q_end = q_start + c3.length - 1;
+                            h_end = h_start + c1.length - 1;
+
+                            var alnstr = '<div STYLE="font-family: monospace;  white-space: pre;">';
+                            alnstr += paddingx(q_start) + ' ' + p3 + ' ' + q_end + '</br>';
+                            alnstr += empty_space + p2 + '</br>';
+                            alnstr += paddingx(h_start) + ' ' + p1 + ' ' + h_end + '</br>';
+                            alnstr += '</font></div></br>';
+                            al.append(alnstr);
+                            i = end;
+                        }
+
+                    });
+                };
+
+                //text alignment tab and use of formatter function to add to the content of the tab
+
+                var al = $('#' + pref + 'alignments');
+                //var hits = data.BlastOutput_iterations.Iteration[0].Iteration_hits.Hit;
+                hits.forEach(function (hit) {
+                    formatter(hit, al);
+                });
+
+            };
+
+            container.empty();
+            container.append("<div><img src=\"" + self.loading_image + "\">&nbsp;&nbsp;loading data...</div>");
+
+            kbws.get_objects([{ref: self.ws_name + "/" + self.ws_id}], function (data) {
+                    ready(data)
+                },
+                function (data) {
+                    container.empty();
+                    container.append('<p>[Error] ' + data.error.message + '</p>');
+                });
+            return this;
+        },
+
+        getData: function () {
+            return {
+                type: "NarrativeTempCard",
+                id: this.ws_name + "." + this.ws_id,
+                workspace: this.ws_name,
+                title: "Temp Widget"
+            };
+        },
+
+        loggedInCallback: function (event, auth) {
+            this.token = auth.token;
+            this.render();
+            return this;
+        },
+
+        loggedOutCallback: function (event, auth) {
+            this.token = null;
+            this.render();
+            return this;
+        },
+        uuid: function () {
```

Comment:
There's a utility library that handles UUIDs now. You can include `util/String` as a dependency, and that provides the `uuid` function. There's an example in the docs here: https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#utilstring